src/dict/stardict.cc

```diff
@@ -1733,27 +1739,27 @@ static void handleIdxSynFile( string const & fileName,
       // punctuation folding. Hopefully there are not a whole lot of valid
       // synonyms which really start from slash and contain dollar signs, or
       // end with dollar and contain slashes.
-      if ( *word == '/' ) {
-        if ( strchr( word, '$' ) )
+      if ( headword.front() == '/' ) {
```

Comment:
This and probably other lines may crash if `wordLen = 0`.

The `wordLen` comes from `strlen(ptr)` many lines before.

Maybe there should be an early `break` of this big for loop if `wordLen = 0`.

The next `ptr` is just `ptr += wordLen + 1;`.

For example, if the data file has two adjacent `\0`, what might happen in this region of code is questionable.

(I only skimmed this function briefly, so I don't know if this is really a case 😅)

Reply:
> for loop if `wordLen = 0`.

should be ok.   this line used in parse the stardict dictionary.

Crash happens when doing the full-text indexing.   

I happened to notice that crash has more great probability to occur when multiple stardict dictionaries in the full-text indexing process at the same time

Reply:
> For example, if the data file has two adjacent `\0`, what might happen in this region of code is questionable.

If this happens ,it is really problemic.
Though We have to rely on the stardict format